external-crates/move/crates/move-compiler/src/parser/syntax.rs

```diff
@@ -243,25 +265,53 @@ fn adjust_token(tokens: &mut Lexer, end_token: Tok) {
     }
 }
 
+fn advance_record_diag(context: &mut Context) {
+    if let Err(diag) = context.tokens.advance() {
+        context.env.add_diag(*diag);
+    }
+}
+
 // Parse a comma-separated list of items, including the specified starting and
 // ending tokens.
 fn parse_comma_list<F, R>(
     context: &mut Context,
     start_token: Tok,
     end_token: Tok,
+    item_first_set: &TokenSet,
     parse_list_item: F,
     item_description: &str,
-) -> Result<Vec<R>, Box<Diagnostic>>
+) -> Vec<R>
 where
     F: Fn(&mut Context) -> Result<R, Box<Diagnostic>>,
 {
     let start_loc = context.tokens.start_loc();
-    consume_token(context.tokens, start_token)?;
+    let at_start_token = context.tokens.at(start_token);
+    if let Err(diag) = consume_token(context.tokens, start_token) {
+        if at_start_token {
+            // advance token past the starting one but something went wrong, still there is a chance
+            // parse the rest of the list
+            let at_stop_set = skip_comma_list_item(context, start_token, end_token, *diag);
+            if at_stop_set {
+                // nothing else to do
+                return vec![];
+            }
+            if context.tokens.at(end_token) {
+                // at the end of the list - consume end token and keep parsing at the outer level
+                advance_record_diag(context);
+                return vec![];
+            }
+        } else {
+            // not even starting token is present - parser has nothing much to do
+            context.env.add_diag(*diag);
+            return vec![];
+        }
```

Comment:
If we are going to use `return`, we should keep really lean into using `return`? 
```suggestion
        if !at_start_token {
            // not even starting token is present - parser has nothing much to do
            context.env.add_diag(*diag);
            return vec![];
        }
        // advance token past the starting one but something went wrong, still there is a chance
        // parse the rest of the list
        let at_stop_set = skip_comma_list_item(context, start_token, end_token, *diag);
        if at_stop_set {
            // nothing else to do
            return vec![];
        }
        if context.tokens.at(end_token) {
            // at the end of the list - consume end token and keep parsing at the outer level
            advance_record_diag(context);
            return vec![];
        }
```